### PR TITLE
Adds global OperationsUseWait which switches Operation.Get to Operation.Wait; manual e2e testing

### PR DESCRIPTION
**Pull request by bowei**
Adds global OperationsUseWait which switches Operation.Get to Operation.Wait

This flag is temporary. After soaking testing, we should be able to remove the Get() path.

Also adds some manual e2e tests in the e2e/ directory.

### CODE REVIEW

File: e2e/observe_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	wantCK := cloud.CallContextKey{
+		Operation: "Get",
+		Version:   meta.VersionAlpha,
+		Service:   "Zones",
+	}
+
+	o.startCK.ProjectID = ""
+	o.startCK.ProjectID = ""
```
Comment (krzykwas) [inline e2e/observe_test.go:66]: It seems you wanted to do

o.startCK.ProjectID = ""
o.endCK.ProjectID = ""

instead?

File: e2e/addresses_test.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			found = true
+		}
+	}
+
+	if !found {
+		t.Fatalf("Expected to find Address %q but it didn't exist", addr1.Name)
+	}
+
+	// Delete
+	err = theCloud.Addresses().Delete(context.Background(), addr1Key)
```
Comment (krzykwas) [inline e2e/addresses_test.go:80]: You can use the ctx defined at line 31.

File: e2e/addresses_test.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+		AddressType: "EXTERNAL",
+		Description: "k8s-cloud-provider-test",
+		Name:        resourceName("addr1"),
+		NetworkTier: "PREMIUM",
+	}
+	addr1Key := meta.GlobalKey(addr1.Name)
+
+	t.Logf("addr1 = %s", fmt.Sprint(addr1))
+
+	t.Cleanup(func() { theCloud.GlobalAddresses().Delete(context.Background(), addr1Key) })
```
Comment (krzykwas) [inline e2e/addresses_test.go:103]: ctx

File: e2e/addresses_test.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+	a, err := theCloud.GlobalAddresses().Get(ctx, addr1Key)
+	if err != nil {
+		t.Fatalf("GlobalAddresses.Get(addr1) = %v", err)
+	}
+	if a.Name != addr1.Name || a.Description != addr1.Description {
+		t.Fatalf("GlobalAddresses.Get() did not match, got %s\nwant %s", pretty.Sprint(a), pretty.Sprint(addr1))
+	}
+
+	// Delete
+	err = theCloud.GlobalAddresses().Delete(context.Background(), addr1Key)
```
Comment (krzykwas) [inline e2e/addresses_test.go:121]: ctx

File: e2e/testmain_test.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+	gerr, ok := err.(*googleapi.Error)
+	if !ok {
+		t.Fatalf("%s: invalid error type, want *googleapi.Error, got %T", fmt.Sprintf(fmtStr, args...), err)
+	}
+	if gerr.Code != wantCode {
+		t.Fatalf("%s: got code %d, want %d (err: %v)", fmt.Sprintf(fmtStr, args...), gerr.Code, wantCode, err)
+	}
+}
+
+// deleteAllResources attempts to remove all resources that may have
```
Comment (krzykwas) [inline e2e/testmain_test.go:120]: Not used, should it be deleted?

File: pkg/cloud/op.go [hunk 6]
```diff
@@ -64,19 +71,36 @@ func (o *gaOperation) isDone(ctx context.Context) (bool, error) {
...
-		klog.V(5).Infof("GA.RegionOperations.Get(%v, %v, %v) = %+v, %v; ctx = %v", o.projectID, o.key.Region, o.key.Name, op, err, ctx)
-	case meta.Zonal:
-		op, err = o.s.GA.ZoneOperations.Get(o.projectID, o.key.Zone, o.key.Name).Context(ctx).Do()
-		klog.V(5).Infof("GA.ZoneOperations.Get(%v, %v, %v) = %+v, %v; ctx = %v", o.projectID, o.key.Zone, o.key.Name, op, err, ctx)
-	case meta.Global:
-		op, err = o.s.GA.GlobalOperations.Get(o.projectID, o.key.Name).Context(ctx).Do()
-		klog.V(5).Infof("GA.GlobalOperations.Get(%v, %v) = %+v, %v; ctx = %v", o.projectID, o.key.Name, op, err, ctx)
-	default:
-		return false, fmt.Errorf("invalid key type: %#v", o.key)
+	if OperationsUseWait {
```
[the remark this answers is not in the document]
Comment (briantkennedy) [reply pkg/cloud/op.go:74]: One of them is not like the other.  In any case, we'll switch to the `Wait` codepath and remove the old one if this works out.

@bowei can you add a comment about cleaning up the old codepath once it's had more time to bake?

**Comment by bowei** [comment]
Updated

**Review by briantkennedy — APPROVED** [review]

**Pull request merged**